### ISSUE TITLE
STIR incorrect spacing for either nii or 2d images

### ISSUE THREAD

**Issue by rijobro** [CCPPETMR/SIRF#562]
```
a = sirf.Reg.NiftiImageData('T1_mf0.nii')
a.get_voxel_sizes()[1:4]

>> array([2.08626, 2.08626, 2.03125], dtype=float32)

a = sirf.STIR.ImageData('uMap_mf0.nii')
a.voxel_sizes()

>> (1.0, 2.08626, 2.08626)
```

For a 2D `.nii` image, `sirf.STIR` incorrectly states that the z-spacing is 1mm.

Things for debugging:

1. use `stir_math` to convert to .hv. What is the reported z-spacing?
2. use an `.nii` image with num voxels in z-direction as 2. Then what is the reported z-spacing?
3. what happens when num voxels in x or y is 1?

**Comment by rijobro** [comment]
I believe this is a STIR issue and will be fixed in SIRF once resolved there. 

STIR issue: https://github.com/UCL/STIR/issues/456.

**Comment by rijobro** [comment]
In the end, I think that this is not really a bug, it's just that `ITK` and `nifti_io` use different norms.
According to `nifti_io`, an image could be 2D, with z-spacing != 1. Conversely, `ITK` will set z-spacing to 1 if the image is 2D. Since STIR uses ITK, we run into this problem. Personally, I don't agree with ITK's logic, but there's not much we can do about that. 

Therefore, upon saving a file from SIRF.Reg, check if the spacing is != 1. If that is the case, increase the dimensionality (e.g., 2D image with z-spacing != 1 is set as a 3D image).

Fixed in df48e5b5daa83bccf879994a93e15b2bd4edf04c.

**Comment by KrisThielemans** [comment]
thanks. I agree that the ITK convention is a bit weird. you could raise an ITK issue in your spare time...
